test/cpp/c10d/FileStoreTest.cpp

```diff
@@ -59,6 +59,24 @@ void testGetSet(std::string path, std::string prefix = "") {
     c10d::test::check(store, "key0", "value0");
     c10d::test::compareSet(store, "key0", "value0", "newValue");
     c10d::test::check(store, "key0", "newValue");
+
+    // Check deleteKey
+    c10d::test::deleteKey(store, "key1");
+    numKeys = fileStore->getNumKeys();
+    EXPECT_EQ(numKeys, 2);
+    c10d::test::check(store, "key0", "newValue");
+    c10d::test::check(store, "key2", "value2");
+
+    c10d::test::set(store, "-key0", "value-");
```

Comment:
I added this test to double-check that keys starting with `-` are not messed up with this PR.